### PR TITLE
cleanup(spanner): use `Status` factory functions

### PR DESCRIPTION
**Pull request by alevenberg**
https://github.com/googleapis/google-cloud-cpp/issues/14107

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14210)

### CODE REVIEW

File: google/cloud/spanner/timestamp.cc [hunk 1]
```diff
@@ -25,7 +26,7 @@ GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 namespace {
 
 Status OutOfRange(std::string message) {
-  return Status(StatusCode::kOutOfRange, std::move(message));
+  return internal::OutOfRangeError(std::move(message), GCP_ERROR_INFO());
```
Comment (coryan) [inline google/cloud/spanner/timestamp.cc:29]: In this case it seems like we would want the caller to give us the error info.

**Review by devbww — APPROVED** [review]

File: google/cloud/spanner/timestamp.cc (hunk 1 above)
Comment (alevenberg) [reply google/cloud/spanner/timestamp.cc]: Done

**Comment by alevenberg** [comment]
PR is ready for review

File: google/cloud/spanner/samples/postgresql_samples.cc [hunk 2]
```diff
@@ -165,10 +166,10 @@ void DmlGettingStartedUpdate(google::cloud::spanner::Client client) {
         auto budget1 = get_budget(txn, 1, 1);
         if (!budget1) return std::move(budget1).status();
         if (*budget1 < transfer_amount) {
-          return google::cloud::Status(
-              google::cloud::StatusCode::kUnknown,
+          return google::cloud::internal::UnknownError(
```
Comment (dbolduc) [inline google/cloud/spanner/samples/postgresql_samples.cc:169]: we shouldnt use internal helpers in public samples

**Review by dbolduc — CHANGES_REQUESTED** [review]

File: google/cloud/spanner/samples/postgresql_samples.cc (hunk 2 above)
Comment (alevenberg) [reply google/cloud/spanner/samples/postgresql_samples.cc]: Done

**Review by dbolduc — APPROVED** [review]

**Pull request merged**